ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -59,8 +59,41 @@ impl InternalIndexerDBService {
         Arc::clone(&self.db_indexer)
     }
 
-    pub async fn run(&mut self) {
-        let mut start_version = self.db_indexer.get_persisted_version().unwrap_or(0);
+    // If the main DB is opened and statekey index is on, we should panic since we will miss all statekeys during the fast sync bootstrap
+    // If node use fast sync, we should start from the fast synced version
+    pub fn get_start_version(&self, node_config: &NodeConfig) -> Version {
+        let indexer_version = self.db_indexer.get_persisted_version().unwrap_or(0);
+        let is_statekeys_empty = self.db_indexer.is_statekeys_empty().unwrap_or(true);
+        let db_min_version = self
+            .db_indexer
+            .get_main_db_lowest_viable_version()
+            .unwrap_or(0);
+        // internal indexer started after the state sync
+        if is_statekeys_empty
+            && db_min_version > 0
+            && self.db_indexer.statekeys_enabled()
+            && node_config
+                .state_sync
+                .state_sync_driver
+                .bootstrapping_mode
+                .is_fast_sync()
```

Comment:
I don't understand this condition .. 
why is it okay when `.is_fast_sync()` is false?

Reply:
rewrote